tests/src/PHPUnit/PhpCodeTest.php

```diff
@@ -7,6 +7,26 @@
 
 class PhpCodeTest extends \PHPUnit_Framework_TestCase
 {
+    public function testMakePhpConstantName()
+    {
+        $this->assertSame('A', PhpCode::makePhpConstantName('a'));
+        $this->assertSame('ABC', PhpCode::makePhpConstantName('abc'));
+        $this->assertSame('A_B_C', PhpCode::makePhpConstantName('a*b##c'));
+        $this->assertSame('ABC', PhpCode::makePhpConstantName('_abc_'));
+        $this->assertSame('ABC', PhpCode::makePhpConstantName('__abc__'));
+        $this->assertSame('AB_C', PhpCode::makePhpConstantName('__AB__c__'));
+
+        // Keyword would be invalid.
+        $this->assertSame('_AS', PhpCode::makePhpConstantName('as'));
+
+        // Leading numeric would be invalid.
+        $this->assertSame('_1A', PhpCode::makePhpConstantName('1A'));
```

Comment:
This assertion is failing with actual result `CONST_1_A`. I don't have a strong opinion whether there should be `CONST_` prefix or not, you can change either test or implementation.